workflow-controller/controller-service/src/main/java/io/hyscale/controller/commands/deploy/HyscaleDeployServiceCommand.java

```diff
@@ -112,78 +128,93 @@
     private DeployComponentInvoker deployComponentInvoker;
     
     @Autowired
-    private ServiceSpecMapper serviceSpecMapper;
-
+    private HyscaleInputSpecProcessor hyscaleInputSpecProcessor;
+    
+    @Autowired
+    private PostValidatorsProvider postValidatorsProvider;
+    
+    @Autowired
+    private WorkflowContextBuilder workflowContextBuilder;
+    
     @Override
     public Integer call() throws Exception {
+        WorkflowLogger.header(ControllerActivity.PROCESSING_INPUT);
         if (!CommandUtil.isInputValid(this)) {
             return ToolConstants.INVALID_INPUT_ERROR_CODE;
         }
+        HyscaleCommandSpecBuilder commandSpecBuilder = new HyscaleCommandSpecBuilder();
+        commandSpecBuilder.setAppName(appName);
+        commandSpecBuilder.setServiceSpecFiles(serviceSpecs);
         
-        Map<String, File> serviceProfileMap = new HashMap<String, File>();
-        try {
-            serviceProfileMap = ServiceProfileUtil.getServiceProfileMap(profiles);
-        } catch (HyscaleException e) {
-            WorkflowLogger.error(ControllerActivity.CANNOT_PROCESS_SERVICE_PROFILE, e.getMessage());
-            throw e;
+        if (profileArg != null) {
+            commandSpecBuilder.setProfileFiles(profileArg.getProfiles());
+            commandSpecBuilder.setProfileName(profileArg.getProfileName());
```

Comment:
We discussed that this object does not contain profiles and name together. 
It could be more resonable if it has some kind ProfileProcessor which return list of profile files.